shared/general/config-json-fields.md

```diff
@@ -0,0 +1,118 @@
+## Valid fields
+
+The following fields can be used to modify the behavior of the host OS.
+
+### hostname
+
+(string) The configured hostname of the device, otherwise the device UUID is used.
+
+### persistentLogging
+
+(boolean) Enable or disable persistent logging on the device - defaults to false.
+
+### country
+
+(string) [Two-letter country code][country-codes] for the country in which the device is operating. This is used for setting the WiFi regulatory domain, and you should check the WiFi device driver for a list of supported country codes.
+
+### ntpServers
+
+(string) A space-separated list of NTP servers to use for time synchronization. Defaults to `resinio.pool.ntp.org` servers:
+
+- `0.resinio.pool.ntp.org`
+- `1.resinio.pool.ntp.org`
+- `2.resinio.pool.ntp.org`
+- `3.resinio.pool.ntp.org`
+
+### dnsServers
+
+(string) A space-separated list of preferred DNS servers to use for name resolution.
+
+- When `dnsServers` is not defined, or empty, Google's DNS server (8.8.8.8) is added to the list of DNS servers obtained via DHCP or statically configured in a NetworkManager connection profile.
```

Comment:
It actually uses both the Google primary and secondary servers (`8.8.8.8` and `8.8.4.4`), see a `/etc/resolv.dnsmasq` config file on-device.

Reply:
Just checking this as this info came from @alexgg comment above https://github.com/balena-io/docs/pull/1249#discussion_r354797800. Specifically `And /run/dnsmasq.servers is the file that is configured from config.json with either the user provided servers or 8.8.8.8.` 

On my sample device, this appears to be true and `8.8.4.4` is not added.

```
root@3a628fc:~# cat /run/dnsmasq.servers

server=8.8.8.8
```

Reply:
OK, that's interesting, maybe later OS changes have changed this (I looked on a 2.38).

Reply:
I think it's this line? https://github.com/balena-os/meta-balena/blob/a7c5f47ef43c9ef79b770ec6ac83989130f5f4ed/meta-balena-common/recipes-support/resin-vars/resin-vars/resin-vars#L90